src/drivers/qnx/lv_qnx.c

```diff
@@ -0,0 +1,341 @@
+    lv_display_set_driver_data(disp, dsc);
+    if(!window_create(disp)) {
+        lv_free(dsc);
+        return NULL;
+    }
+
+    lv_display_set_flush_cb(disp, flush_cb);
+
+    if(!init_display_from_window(disp)) {
+        screen_destroy_window(dsc->window);
+        lv_free(dsc);
+        return NULL;
+    }
+
+    return disp;
+}
+
+void lv_qnx_window_set_title(lv_display_t * disp, const char * title)
+{
+    lv_qnx_window_t * dsc = lv_display_get_driver_data(disp);
+    screen_event_t event;
+    screen_create_event(&event);
+
+    char title_buf[64];
+    snprintf(title_buf, sizeof(title), "Title=%s", title);
+    screen_set_event_property_iv(event, SCREEN_PROPERTY_TYPE,
+    (const int[]) {
+        SCREEN_EVENT_MANAGER
+    });
+    screen_set_event_property_cv(event, SCREEN_PROPERTY_USER_DATA,
+                                 sizeof(title_buf), title_buf);
+    screen_set_event_property_pv(event, SCREEN_PROPERTY_WINDOW,
+                                 (void **)&dsc->window);
+    screen_set_event_property_pv(event, SCREEN_PROPERTY_CONTEXT,
+                                 (void **)&context);
+
+    screen_inject_event(NULL, event);
+}
+
+bool lv_qnx_inputs_create(lv_display_t * disp)
+{
+    lv_indev_t * indev = lv_indev_create();
+    lv_indev_set_type(indev, LV_INDEV_TYPE_POINTER);
+    lv_indev_set_read_cb(indev, get_pointer);
+    return true;
+}
+
+int lv_qnx_event_loop(lv_display_t * disp)
+{
+    /*Run the event loop*/
+    screen_event_t  event;
+    if(screen_create_event(&event) != 0) {
+        perror("screen_create_event");
+        return EXIT_FAILURE;
+    }
+
+    for(;;) {
+        /*Wait for an event, timing out after 16ms if animations are running*/
+        uint64_t timeout = lv_anim_count_running() > 0  ? 16000000UL : -1UL;
+        if(screen_get_event(context, event, timeout) != 0) {
+            perror("screen_get_event");
+            return EXIT_FAILURE;
+        }
+
+        /*Get the event's type*/
+        int type;
+        if(screen_get_event_property_iv(event, SCREEN_PROPERTY_TYPE, &type)
+           != 0) {
+            perror("screen_get_event_property_iv(TYPE)");
+            return EXIT_FAILURE;
+        }
+
+        if(type == SCREEN_EVENT_POINTER) {
+            /*Handle a mouse pointer event*/
+            if(screen_get_event_property_iv(event,
+                                            SCREEN_PROPERTY_SOURCE_POSITION,
+                                            mouse_pos)
+               != 0) {
+                perror("screen_get_event_property_iv(SOURCE_POSITION)");
+                return EXIT_FAILURE;
+            }
+
+            if(screen_get_event_property_iv(event, SCREEN_PROPERTY_BUTTONS,
+                                            &mouse_buttons)
+               != 0) {
+                perror("screen_get_event_property_iv(BUTTONS)");
+                return EXIT_FAILURE;
+            }
+        }
+
+        lv_timer_handler();
```

Comment:
`lv_timer_handler` returns the time for next event. Use it as the timeout.

Reply:
I tried that, but I don't see the call ever returning `LV_NO_TIMER_READY`, even when the application is idle:

```
        /*Calculate the next timeout*/
        uint32_t timeout_ms = lv_timer_handler();
        if(timeout_ms == LV_NO_TIMER_READY) {
            timeout_ns = -1ULL;
        } else {
            timeout_ns = (uint64_t)timeout_ms * 1000000UL;
        }
```

Reply:
`LV_NO_TIMER_READY ` should be regarded as an error case that LVGL has nothing more to do.
The `timeout_ms` returned is the time before LVGL needs to take action.

The example of NuttX porint is more clear to explain it.

https://github.com/lvgl/lvgl/blob/48ce05d71e782eb149a3ea41e05cb64f6270ffa7/src/drivers/nuttx/lv_nuttx_libuv.c#L124-L143

Reply:
Thanks!
Is it expected that, when idle, `lv_timer_handler()` returns 33ms? That's what I am observing. I have no problem with making the change if this is the expected behaviour, but I am uncomfortable with an idle process waking up at such a frequency.

Reply:
OK, I see where it's coming from. It's the indev timer, with `lv_indev_read_timer_cb` as the callback. I need a way to inject input events rather than store them and then have them read periodically.

Reply:
Fixed, but now the display refresh timer is waking up every 33ms, even when there is no animation running.

Reply:
I see that it's working, but it still might help: [Docs about using indevs in event driven mode](https://docs.lvgl.io/master/porting/indev.html#switching-the-input-device-to-event-driven-mode)

Have you enabled `LV_USE_PERF_MONITOR` or `LV_USE_MEM_MONITOR`? They also keep the display timer running.